Chatkit/Auth/TokenProvider.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TokenProvider.swift
+//  Chatkit
+//
+//  Created by Mike Pye on 06/12/2019.
+//  Copyright © 2019 Pusher Ltd. All rights reserved.
+//
+// STUB FILE. This file demonstrates the desired interface, but contains to implementation
+// TODO: Move many of these types to PusherPlatform
+
+import Foundation
+import PusherPlatform
+
+// TODO: Temporary
+extension String: Error { }
+
+// Re-export so that PusherPlatform import is not required in client code
+public typealias TokenProvider = PPTokenProvider
+public typealias TokenProviderResult = PPTokenProviderResult
```

Comment:
Improving the platform lib is not in the scope? Wouldn't it be better to give them better names at the source? `P` prefix is not very clear. I think that `Chatkit` prefix will be the best or `Pusher` if it's something not only specific to `Chatkit`.

Regarding re-exporting couldn't it be handled here like `api` dependency in Gradle?

Reply:
Renaming everything is the platform library not to have a `PP` prefix (which is an Obj-C convention we want to drop) is a possibility, but not something I want to dive in to right now.
Improvements are in scope though.

The re-export is because swift imports work quite differently to android - they aren't automatic, and they are whole module, not per-type. So I don't want users of the SDK to have to `import PusherPlatform` at all to use the SDK.

Reply:
2-3 letter prefixes are Objective-C way of introducing a fake namespace (2 letters for Apple frameworks, 3 letters for third parties). Swift creates namespaces using modules defied in a `modulemap` file. I am not sure if that is a concern but it is possible for us to provide different names for Objective-C and Swift.

It is possible in Swift (and Objective-C) to import a single type:

```
import class PusherPlatform.Instance
```

Reply:
Oh, that's cool. I haven't seen anyone use it!